controllers/apply/contacts-next/fields.js

```diff
@@ -0,0 +1,1829 @@
+'use strict';
+const flatMap = require('lodash/flatMap');
+const get = require('lodash/fp/get');
+const moment = require('moment');
+const { oneLine } = require('common-tags');
+
+const Joi = require('../lib/joi-extensions');
+
+const EmailField = require('../lib/field-types/email');
+const DateField = require('../lib/field-types/date');
+const PhoneField = require('../lib/field-types/phone');
+const NameField = require('../lib/field-types/name');
+
+const fieldContactLanguagePreference = require('./fields/contact-language-preference');
+const fieldOrganisationStartDate = require('./fields/organisation-start-date');
+const fieldOrganisationType = require('./fields/organisation-type');
+const fieldProjectCountry = require('./fields/project-country');
+const fieldProjectTotalCosts = require('./fields/project-total-costs');
+const fieldSeniorContactRole = require('./fields/senior-contact-role');
+const fieldTotalIncomeYear = require('./fields/total-income-year');
+const fieldYourIdeaCommunity = require('./fields/your-idea-community');
+const fieldYourIdeaPriorities = require('./fields/your-idea-priorities');
+const fieldYourIdeaProject = require('./fields/your-idea-project');
+const fieldProjectLocation = require('./fields/project-location');
+const fieldProjectDateRange = require('./fields/project-date-range');
+const fieldProjectStartDate = require('./fields/project-start-date');
+const fieldProjectEndDate = require('./fields/project-end-date');
+const fieldCompanyNumber = require('./fields/company-number');
+const fieldCharityNumber = require('./fields/charity-number');
+const fieldEducationNumber = require('./fields/education-number');
+const fieldAddress = require('./fields/address');
+const fieldBankStatement = require('./fields/bank-statement');
+const fieldBuildingSocietyNumber = require('./fields/building-society-number');
+const fieldBankAccountNumber = require('./fields/bank-account-number');
+const fieldBankSortCode = require('./fields/bank-sort-code');
+const fieldBankAccountName = require('./fields/bank-account-name');
+
+const {
+    BENEFICIARY_GROUPS,
+    CONTACT_EXCLUDED_TYPES,
+    MAX_BUDGET_TOTAL_GBP,
+    MIN_AGE_MAIN_CONTACT,
+    MIN_AGE_SENIOR_CONTACT,
+    MIN_BUDGET_TOTAL_GBP,
+    ORGANISATION_TYPES,
+    STATUTORY_BODY_TYPES,
+    FREE_TEXT_MAXLENGTH
+} = require('./constants');
+
+module.exports = function fieldsFor({ locale, data = {}, flags = {} }) {
+    const localise = get(locale);
+
+    const seniorContactName = [
```

Comment:
These lines here are new – several fields now need access to one of the contact names.